### PR TITLE
lib: pdn rework

### PR DESCRIPTION
**Pull request by lemrey**
Rework `pdn_managment` library and rename it to simply `pdn`.

- Removed `pdn_init_and_connect` and `pdn_disconnect`
These where just too thin, and provided no functionality on top of socket APIs.

- Added `pdn_activate` which will create and connect a PDN socket, if necessary.

FYI: @metc

### CODE REVIEW

File: include/pdn_management.h [hunk 1]
```diff
@@ -13,21 +13,18 @@
...
+ * @param[in,out]	fd	The socket handle.
+ * @param[in]		apn	The packet data network name.
  *
- * @param[in] pdn_fd Identifies the PDN for which the procedure is requested.
+ * @return 0 if PDN socket was valid and already connected.
+ * @return 1 if PDN socket has been recreated.
+ * @return -1 on error.
  */
-void pdn_disconnect(int pdn_fd);
+int pdn_activate(int *fd, char *apn);
```
Comment (metc) [inline include/pdn_management.h]: Please add `const` qualifiers

Comment (lemrey) [reply include/pdn_management.h]: Done

**Comment by lemrey** [comment]
Yes @SebastianBoe, we should add this to the changelog before merging.

File: lib/pdn/pdn.c [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+ */
+
+#include <stddef.h>
+#include <net/socket.h>
+#include <nrf_socket.h>
+
+int pdn_activate(int *fd, const char *apn)
+{
+	int err;
+	nrf_pdn_state_t active = 0;
```
Comment (rlubos) [inline lib/pdn/pdn.c]: Can we use `u8_t`/`socklen_t` to avoid `nrf_*` namespace usage?

File: lib/pdn/CMakeLists.txt [hunk 3]
```diff
@@ -5,4 +5,4 @@
 #
 
 zephyr_library()
-zephyr_library_sources(pdn_management.c)
\ No newline at end of file
+zephyr_library_sources(pdn.c)
```
Comment (rlubos) [inline lib/pdn/CMakeLists.txt]: Missing new line.

File: lib/pdn/Kconfig [hunk 4]
```diff
@@ -4,6 +4,6 @@
 # SPDX-License-Identifier: LicenseRef-BSD-5-Clause-Nordic
 #
 
-menuconfig PDN_MANAGEMENT
-	bool "PDN Management for nRF9160"
+config PDN
+	bool "PDN library"
 	depends on BSD_LIBRARY
```
Comment (rlubos) [inline lib/pdn/Kconfig]: Missing new line.

**Pull request closed without merging**